src/dhcpmon/src/dhcp_mon.c

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file dhcp_mon.c
+ *
+ *  @brief dhcp relay monitor module
+ */
+
+#include <signal.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <syslog.h>
+#include <assert.h>
+
+#include "dhcp_mon.h"
+#include "dhcp_devman.h"
+
+/** window_interval_sec monitoring window for dhcp relay health checks */
+static int window_interval_sec = 12;
+/** dhcp_unhealthy_max_count max count of consecutive unhealthy statuses before reporting to syslog */
+static int dhcp_unhealthy_max_count = 10;
+/** dhcp_mon_active control monitoring thread main loop */
+static int dhcp_mon_active = 0;
+
+/**
+ * @code dhcp_mon_process(arg);
+ *
+ * @brief monitoring thread main loop. It creates a timer that delivers SIGALRM every window_sec seconds.
+ *        when signal is delivered, dhcp relay status is queried from devman.
+ *
+ * @param arg pointer to user argument
+ *
+ * @return pointer to exist status
+ */
+static void* dhcp_mon_process(void *arg)
+{
+    int rv;
+    struct itimerspec ts;
+    sigset_t sigset;
+    struct sigevent se;
+    timer_t timer_id;
+    static int count = 0;
+
+    sigemptyset(&sigset);
+    sigaddset(&sigset, SIGALRM);
+
+    /* setup timer */
+    ts.it_interval.tv_sec = window_interval_sec;
+    ts.it_interval.tv_nsec = 0;
+    ts.it_value.tv_sec = window_interval_sec;
+    ts.it_value.tv_nsec = 0;
+
+    se.sigev_value.sival_ptr = NULL;  // data passed with the notification
+    se.sigev_signo = SIGALRM;
+    se.sigev_notify = SIGEV_THREAD_ID;
+    se._sigev_un._tid = syscall(__NR_gettid);
```

Comment:
Prefer passing `SYS_gettid` instead of `__NR_gettid`

Reply:
code will be remove in the upcoming patch